tests/test_browser.py

```diff
@@ -1076,6 +1076,53 @@ def test_sdl_joystick_2(self):
     Popen([PYTHON, EMCC, os.path.join(self.get_dir(), 'sdl_joystick.c'), '-O2', '--minify', '0', '-o', 'page.html', '--pre-js', 'pre.js', '-lSDL', '-lGL']).communicate()
     self.run_browser('page.html', '', '/report_result?2')
 
+  def test_glfw_joystick(self):
```

Comment:
Great to have a mock input injecting test! It would be nice to add the same test to the `interactive` suite as well, but without the API mock implementation part, so that could be tested with a live gamepad.